src/web/context/GlobalContext.tsx

```diff
@@ -0,0 +1,231 @@
+// React
+import React, { createContext, ReactNode, useEffect, useReducer } from 'react'
+
+// Jaia
+import { jaiaAPI } from '../jcc/common/JaiaAPI'
+
+export interface GlobalContextType {
+    clientID: string,
+    controllingClientID: string
+    selectedPodElement: SelectedPodElement
+    showHubDetails: boolean
+    hubAccordionStates: HubAccordionStates
+    isRCMode: boolean
+}
+
+interface SelectedPodElement {
+    type: PodElement
+    id: number
+}
+
+interface HubAccordionStates {
+    quickLook: boolean,
+    commands: boolean,
+    links: boolean
+}
+
+interface Action {
+    type: string,
+    clientID?: string,
+    hubAccordionName?: string
+}
+
+interface GlobalContextProviderProps {
+    children: ReactNode
+}
+
+export enum PodElement {
+    'BOT' = 1,
+    'HUB' = 2
+}
+
+const defaultHubAccordionStates = {
+    quickLook: false,
+    commands: false,
+    links: false
+}
+
+export const globalDefaultContext: GlobalContextType = {
+    clientID: '',
+    controllingClientID: '',
+    selectedPodElement: null,
+    showHubDetails: false,
+    hubAccordionStates: defaultHubAccordionStates,
+    isRCMode: false
+}
+
+export const GlobalContext = createContext(null)
+export const GlobalDispatchContext = createContext(null)
+
+/**
+ * Updates GlobalContext
+ * 
+ * @param {GlobalContextType} state Holds the most recent reference to state 
+ * @param {Action} action Contains data associated with a state update 
+ * @returns {GlobalContextType} A copy of the updated state
+ */
+function globalReducer(state: GlobalContextType, action: Action) {
+    let mutableState = {...state}
+    switch (action.type) {
+        case 'SAVED_CLIENT_ID':
+            return handleSavedClientID(mutableState, action.clientID)
```

Comment:
Should we check if clientID exists

Reply:
It makes me wonder whether the action/reducer pattern is solving the same problem as the delegate pattern solves.  Except that with a delegate object, the actions would be methods, and you'd be able to document, enforce, and type-check all of the parameters for each method.  There may be some difference between the two patterns that I'm not seeing, though...

Reply:
Hmm I think there is some overlap. I think with React we need to stick with the action/reducer pattern since under the hood the reducer updates state with the value it returns. I suppose we could make the action object expect functions, but it might add an extra challenge to maintainability and onboarding since we deviate from the uses shown in the React docs.